### PR TITLE
Add Vale config

### LINKED ISSUE #3586: Add a Vale configuration to Tarantool docs
[Vale](https://vale.sh/) is a command-line tool that checks documentation against predefined rules: grammar, style, terms, and so on.

### PR DESCRIPTION
**Pull request by andreyaksenov**
Added the Vale configuration to the documentation. This configuration includes two predefined packages that implement Microsoft and Google style guides:
- https://vale.sh/hub/microsoft/
- https://vale.sh/hub/google/

Moreover, this PR includes a custom `Tarantool` style, which includes several rules from the [GitLab config](https://gitlab.com/gitlab-org/gitlab/-/tree/master/doc/.vale/gitlab).

Apart from three styles, the `styles/Vocab/Tarantool/accept.txt` file contains valid Tarantool-specific terms.

To show Vale warnings in the PyCharm editor, follow the steps below (macOS):

1. Install Vale: `brew install vale`.
2. Install Sphinx for RST support: `brew install sphinx-doc`.
3. Make sure that the Grazie Professional plugin is installed and enabled.
4. On the `Preferences | Editor | Natural Languages | Grazie Professional` page, turn on the `Enable Vale annotations` option.
5. In the repository folder, execute the `vale sync` command.

PyCharm should now use the Vale config automatically (`.vale.ini`) and show warnings right in the editor:
image

Please add your suggestions regarding customizing Microsoft/Google styles or adding Tarantool-specific rules and terms.

> TODO:
> - Add information about Vale configuration to README.

### CODE REVIEW

File: styles/Tarantool/Latin.yml
Comment (p7nov) [inline styles/Tarantool/Latin.yml:1]: Already added in Foreign.yml

File: styles/Tarantool/WordList.yml
Comment (p7nov) [inline styles/Tarantool/WordList.yml:1]: Maybe join with Terms.yml?

File: styles/Vocab/Tarantool/accept.txt [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+nginx
+serializable
+unserializable
+autocommit
+WAL
+sharding
+sharded
+nullable
+coroutines
+multikey
```
Comment (p7nov) [inline styles/Vocab/Tarantool/accept.txt]:
```suggestion
multikey
datetime
varbinary
```

File: styles/Tarantool/WordList.yml [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  above: preceding
+  account name: username
+  action bar: app bar
+  admin: administrator
+  Ajax: AJAX
+  a\.k\.a|aka: or|also known as
+  Android device: Android-powered device
+  android: Android
+  API explorer: APIs Explorer
+  application: app
```
Comment (p7nov) [inline styles/Tarantool/WordList.yml]: Not sure this is valid for our case.

File: styles/Tarantool/ComplexWords.yml [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+  indication: sign
+  initiate: start|begin
+  itemized: listed
+  jeopardize: risk
+  liaise: work with|partner with
+  maintain: keep|support
+  methodology: method
+  modify: change
+  monitor: check|watch
+  multiple: many
```
Comment (p7nov) [inline styles/Tarantool/ComplexWords.yml]: 'multiple' seems fine to me.

File: styles/Tarantool/ComplexWords.yml [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+  compensate: pay
+  component: part
+  comprise: form|include
+  concept: idea
+  concerning: about
+  confer: give|award
+  consequently: so
+  consolidate: merge
+  constitutes: forms
+  contains: has
```
Comment (p7nov) [inline styles/Tarantool/ComplexWords.yml]: 'contains' seems fine to me.

File: styles/Tarantool/WordList.yml [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+  approx\.: approximately
+  authN: authentication
+  authZ: authorization
+  autoupdate: automatically update
+  cellular data: mobile data
+  cellular network: mobile network
+  chapter: documents|pages|sections
+  check box: checkbox
+  check: select
```
Comment (p7nov) [inline styles/Tarantool/WordList.yml]: Not our case, I think.

File: styles/Tarantool/CurrentStatus.yml [hunk 6]
```diff
@@ -0,0 +1,6 @@
+extends: existence
+message: "Remove '%s'. The documentation reflects the current state of the product."
+level: warning
+ignorecase: true
+tokens:
+  - currently
```
Comment (p7nov) [inline styles/Tarantool/CurrentStatus.yml:6]:
```suggestion
  - currently
  - previously
  - now
  - today
  - in(?: the|) future
```

**Review by p7nov — APPROVED** [review]
Great 👍 
I've added some points I've spotted when checking the rules on random doc pages.

Anyway, we can push this config and adapt it further later.

File: styles/Tarantool/WordList.yml
Comment (andreyaksenov) [reply styles/Tarantool/WordList.yml:1]: Moved `Terms.yml` words to `WordList.yml`.

File: styles/Tarantool/WordList.yml (hunk 2 above)
Comment (andreyaksenov) [reply styles/Tarantool/WordList.yml]: agree, forgot to remove

**Review by xuniq — APPROVED** [review]

**Pull request merged**